ome/ome-model
# Add build support for maven and cmake

Changes:

- Remove ant build files
- Move components/* to the top-level
- Each component uses a standard maven project layout
- Specification schemas and transforms are relocated under `src/main/resources`
- C++ code relocated under `src/main/cpp` and `src/test/cpp` so the C++ and Java code now sits side by side, to make the layout cleaner and more consistent; they can e.g. share resources and test data in the future
- Correct all paths to use the new locations
- xsd-fu module uses pom packaging to prevent it making a jar
- Top-level build is mostly derived from the ome-common-java pom, but with added use of `pluginManagement` for the sub-projects.

To do:

- Release process; we'll need to strip out genshi and the other python bits to make a `-dfsg` release as for bioformats.  The script needs copying over an adapting
- The C++ code is using git for versioning as for the other repos; we could update it to get the version from the toplevel pom to avoid inconsistency.

Testing:

- Build with maven
- Check out https://github.com/rleigh-codelibre/bioformats/tree/ome-model-split
- Build bioformats with ant and maven; it will use the specification and ome-xml jars

For C++:

Create a git tag with e.g. `git tag -m "Dummy release tag" v5.3.9-wip`
Build with the superbuild as normal, but configure with `-Dome-xml-dir=/path/to/ome-model`.  `ome-model` is directly substitutable for `bioformats`.
